app/controllers/api/admin/ldap_connection_test_controller.rb

```diff
@@ -0,0 +1,38 @@
+# encoding: utf-8
+
+#  Copyright (c) 2008-2017, Puzzle ITC GmbH. This file is part of
+#  Cryptopus and licensed under the Affero General Public License version 3 or later.
+#  See the COPYING file at the top-level directory or at
+#  https://github.com/puzzle/cryptopus.
+
+class Api::Admin::LdapConnectionTestController < ApiController
+
+  def new
+    if hostlist.present?
+      hostlist.each do |hostname|
+        message(hostname)
+      end
+    else
+      add_error(t('flashes.api.admin.settings.test_ldap_connection.no_hostname_present'))
+    end
+    render_json ''
+  end
+
+  private
+
+  def message(hostname)
+    ldap_connection.test_connection(hostname)
```

Comment:
make #test_connection returning an collection of successful and failed hosts
e.g. { success: ['server1'], failed: ['server3'] }